spec/ruby/core/tracepoint/inspect_spec.rb

```diff
@@ -31,6 +78,34 @@ class TracePointSpec::C
       end
     end
 
-    inspect.should == "#<TracePoint:class@#{__FILE__}:#{line}>"
+    inspect.should == "#<TracePoint:class #{__FILE__}:#{line}>"
+  end
+
+  it 'returns a String showing the event and thread for :thread_begin event' do
+    inspect = nil
+    thread_inspection = nil
+    TracePoint.new(:thread_begin) { |tp|
+      inspect ||= tp.inspect
+    }.enable do
+      thread = Thread.new {}
+      thread_inspection = thread.inspect
+      thread.join
+    end
+
+    inspect.should == "#<TracePoint:thread_begin #{thread_inspection}>"
+  end
+
+  it 'returns a String showing the event and thread for :thread_end event' do
+    inspect = nil
+    thread_inspection = nil
+    TracePoint.new(:thread_end) { |tp|
+      inspect ||= tp.inspect
+    }.enable do
+      thread = Thread.new {}
+      thread_inspection = thread.inspect
+      thread.join
+    end
+
+    inspect.should == "#<TracePoint:thread_end #{thread_inspection}>"
```

Comment:
As the CI shows: https://travis-ci.org/github/ruby/ruby/jobs/715089657
The specs need to pass on older versions of Ruby too.

So you need to use `ruby_version_is ""..."2.8" do` and `ruby_version_is "2.8" do` guards.

I think a convenient way in this case is to define a local variable at the top of the file (or an `@ivar` in a `before` block) like:
```ruby
sep = "@"
ruby_version_is "2.8" do
  sep = " "
end
```
and use it in the expected result.